### PR TITLE
[Merged by Bors] - post release 0.10.7 notes

### PR DESCRIPTION
**Pull request by digikata**
- chore: post-release release_checklist.md update
- chore: Release.md, add pointer to integrated platform release docs

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Release process
 
+See [Platform Release doc](https://github.com/infinyon/fluvio-cloud-ops/blob/master/docs/release/RELEASE.md) for integrated platform release process steps.
+
```
Comment (morenol) [inline RELEASE.md:4]: maybe we should not mention this here?

Comment (digikata) [reply RELEASE.md:4]: I'm was undecided about this, but it put it in for discussion.  We should only have one release procedure. Maybe we should remove most of this RELEASE file and depend on the fc-ops one? It was nicer to only have to look at one prodcedure (+ issues).

Comment (sehz) [reply RELEASE.md:4]: remove cloud ops release

Comment (digikata) [reply RELEASE.md]: removed!

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**